### ISSUE TITLE
Reset simulator Xcode 9.2+

### ISSUE THREAD

**Issue by danielbowden** [fastlane/fastlane#11572]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Hi,
Just a quick question, what is now the supported and suggested method for erasing the simulator?

I have previously use the `reset_simulator_contents` action but it's source code states it doesn't work for Xcode 9 and above. (https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/reset_simulator_contents.rb#L5)

We are wishing to erase the simulator on our CI between builds/tests.

Thanks!

### Environment

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.12.6                                     |
| Ruby                        | 2.2.2                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.14.3 (Apple Git-98)           |
| Installation Source         | ~/.rbenv/versions/2.2.2/bin/fastlane        |
| Host                        | Mac OS X 10.12.6 (16G29)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.2.2/lib                 |
| OpenSSL Version             | OpenSSL 1.0.2l  25 May 2017                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 9.2                                         |

**Comment by huynguyen** [comment]
I've also run into this. In https://github.com/fastlane/fastlane/pull/10326#issuecomment-329581668 @ohayon mentioned it was because the betas weren't working. Is this still the case? I can create a PR to remove the warning, if that isn't the case anymore.

In the meantime I've used the workaround mentioned in that PR by calling the Fastlane action directly via `FastlaneCore::Simulator.reset_all`.

**Comment by danielbowden** [comment]
Thanks for the reply @huynguyen. I can use that workaround for now but will keep an eye on this for an update on the action.

I seem to remember resetting a particular simulator in the past not all, or all for a iOS version but the `reset_all` and `reset_all_by_version` methods should be fine for our CI machines.

**Comment by danielbowden** [comment]
Haha fastlane bot, you’re right there hasn’t been any activity on this but I’ve still kept the tab open to check for a reply every few days.

**Comment by danielbowden** [comment]
Same comment as my last one above..

In the meantime I'm using the following in my Fastfile but it's not as clean as the previous action.

```ruby
  private_lane :erase_simulator do
    test_device = 'iPhone 7'
    simulators = FastlaneCore::DeviceManager.simulators('iOS')
    # Find latest simulator for test_device
    device = simulators.select { |s| s.name == test_device }
                 .sort_by { |s| Gem::Version.create(s.os_version) }
                 .last

    FastlaneCore::Simulator.reset(udid: device.udid) unless device.nil?
  end
```

**Comment by danielbowden** [comment]
Same comment as my last one above..

**Comment by bes** [comment]
Would love to have this issue fixed as well =)

**Comment by iosdevben** [comment]
Likewise

**Comment by jhandguy** [comment]
Same

**Comment by jhandguy** [comment]
This bug has been fixed in **fastlane 2.100.1**! 🍾
